src/services/test_results/TestResultsBuilder.ts

```diff
@@ -1,64 +1,154 @@
+        let gherkinResults = this.getCucumberReportsFromPath(cucumberReportsPath, logger);
         let buildApi: ba.IBuildApi = await connection.getBuildApi();
         let build = await buildApi.getBuild(projectName, buildId);
         let buildURI = build.uri;
         let testApi: ta.ITestApi = await connection.getTestApi();
+
+        if (gherkinResults.length > 0) {
+            xmlTestResults.push(TestResultsBuilder.getGherkinTestResultXml(gherkinResults, serverId, jobId, buildId.toString(), logger))
+        }
+
+        let processedTests = this.getProcessedTestNames(gherkinResults);
         let testRuns = await testApi.getTestRuns(projectName, buildURI);
+
         if (testRuns.length > 0) {
             let results: TestCaseResult[] = [];
-            for (let i = 0; i < 1; i++) {
+
+            for (let i = 0; i < testRuns.length; i++) {
                 let testRunId = testRuns[i].id;
                 results = results.concat(await testApi.getTestResults(projectName, testRunId));
             }
-            return TestResultsBuilder.getTestResultXml(results, serverId, jobId, logger);
+
+            results = results.filter(result => !processedTests.has(result.automatedTestName))
+
+            if (results.length > 0) {
+                xmlTestResults.push(TestResultsBuilder.getUnitTestResultXml(results, serverId, jobId, logger));
+            }
         } else {
-            logger.info('No test results found');
-            return null;
+            logger.info('No unit test results found');
+        }
+
+        return xmlTestResults;
+    }
+
+    private static getProcessedTestNames(gherkinResults: GherkinResultData[]): Set<string> {
+        let processedTests: Set<string> = new Set();
+
+        gherkinResults.forEach(gherkinRes => {
+            let scenarios: GherkinScenarioData[] = Array.isArray(gherkinRes.feature.scenarios.scenario)
+                ? gherkinRes.feature.scenarios.scenario
+                : Array.of(gherkinRes.feature.scenarios.scenario);
+            scenarios.forEach(scenario => processedTests.add(scenario._attributes.name));
+        });
+
+        return processedTests;
+    }
+
+    private static getCucumberReportsFromPath(cucumberReportsPath: string, logger: LogUtils): GherkinResultData[] {
+        let gherkinResults: GherkinResultData[] = [];
+
+        if (cucumberReportsPath == undefined) {
+            return gherkinResults;
         }
+
+        const inputPath = cucumberReportsPath.replace(/\\/g, '/')
+        const files = globby.sync([`${inputPath}/*.xml`])
+        logger.info(`Found ${files.length} matching ${inputPath} pattern`)
+
+        let options = {compact: true, ignoreComment: true, spaces: 4};
+
+        files.forEach(filePath => {
+            logger.info(`Processing ${filePath}`)
+            const rawContent = readFileSync(filePath, 'utf-8')
+            const featuresAsJson = convert.xml2js(rawContent, options);
+            const features = featuresAsJson.features.feature;
```

Comment:
Why you retrieve features.feature, and the const is called features??

Reply:
Because in XML the features tag can contain multiple feature tags inside it. In that situation the features.feature will be a list containing all of the converted "feature" tags. As the mapping is made one to one, the naming of the generated javascript fields classes is misleading. That is why I named the const features. so that we can know that It can be a list containing multiple features.